pkg/doc/verifiable/credential.go

```diff
@@ -222,13 +253,20 @@ type CredentialDecoder func(dataJSON []byte, credential *Credential) error
 // CredentialTemplate defines a factory method to create new Credential template.
 type CredentialTemplate func() *Credential
 
+// PublicKeyFetcher fetches public key for JWT signing verification based on Issuer ID (possibly DID)
+// and Key ID.
+// If not defined, JWT encoding is not tested.
```

Comment:
It looks like we're always fetching the key if `verifyJwtSignature` is `true`. Maybe remove this line?